201-vmss-master-slave-customscript/azuredeploy.json

```diff
@@ -0,0 +1,466 @@
+   },
+   "resources":[  
+      {  
+         "type":"Microsoft.Network/virtualNetworks",
+         "name":"[variables('vnetName')]",
+         "apiVersion":"2017-06-01",
+         "location":"[parameters('location')]",
+         "properties":{  
+            "addressSpace":{  
+               "addressPrefixes":[  
+                  "10.0.0.0/16"
+               ]
+            },
+            "subnets":[  
+               {  
+                  "name":"[variables('subnetMaster')]",
+                  "properties":{  
+                     "addressPrefix":"10.0.0.0/24"
+                  }
+               },
+               {  
+                  "name":"[variables('subnetData')]",
+                  "properties":{  
+                     "addressPrefix":"10.0.1.0/24"
+                  }
+               }
+            ]
+         }
+      },
+      {  
+         "type":"Microsoft.Network/publicIPAddresses",
+         "name":"[variables('masterPublicIPAddressName')]",
+         "apiVersion":"2017-06-01",
+         "location":"[parameters('location')]",
+         "properties":{  
+            "publicIPAllocationMethod":"Dynamic",
+            "dnsSettings":{  
+               "domainNameLabel":"[concat(variables('vmssDomainName'),'-master-node')]"
+            }
+         }
+      },
+      {  
+         "type":"Microsoft.Network/publicIPAddresses",
+         "name":"[variables('dataPublicIPAddressName')]",
+         "apiVersion":"2017-06-01",
+         "location":"[parameters('location')]",
+         "properties":{  
+            "publicIPAllocationMethod":"Dynamic",
+            "dnsSettings":{  
+               "domainNameLabel":"[concat(variables('vmssDomainName'),'-data-node')]"
+            }
+         }
+      },
+      {  
+         "type":"Microsoft.Network/loadBalancers",
+         "name":"[variables('masterNodeLoadBalancerName')]",
+         "apiVersion":"2017-06-01",
+         "location":"[parameters('location')]",
+         "properties":{  
+            "frontendIPConfigurations":[  
+               {  
+                  "name":"[variables('masterNodeLoadBalancerFrontEndName')]",
+                  "properties":{  
+                     "publicIPAddress":{  
+                        "id":"[resourceId('Microsoft.Network/publicIPAddresses', variables('masterPublicIPAddressName'))]"
+                     }
+                  }
+               }
+            ],
+            "backendAddressPools":[  
+               {  
+                  "name":"[variables('masterNodeLoadBalancerBackendName')]"
+               }
+            ],
+            "loadBalancingRules":[  
+               {  
+                  "name":"roundRobinLBRule",
+                  "properties":{  
+                     "frontendIPConfiguration":{  
+                        "id":"[concat(resourceId('Microsoft.Network/loadBalancers', variables('masterNodeLoadBalancerName')), '/frontendIPConfigurations/', variables('masterNodeLoadBalancerFrontEndName'))]"
```

Comment:
simplify, just use resourceId() - just one comment here but all need to be updated.